ntbs-service/Pages/Notifications/Edit/ClinicalDetails.cshtml.cs

```diff
@@ -312,7 +312,7 @@ public ContentResult OnPostValidateNotificationSiteProperty([FromBody]InputValid
 
         public ContentResult OnPostValidateClinicalDetailsYearComparison([FromBody]YearComparisonValidationModel validationData)
         {
-            return ValidationService.GetYearComparisonValidationResult(validationData.NewYear, validationData.ExistingYear, validationData.PropertyName);
+            return validationData == null ? Content("") : ValidationService.GetYearComparisonValidationResult(validationData.NewYear, validationData.ExistingYear, validationData.PropertyName);
```

Comment:
I don't understand why `validationData` was null here? Surely we were sending a request with a null `NewYear`?

Reply:
I had the same thought yesterday and it piqued my interest and I decided to have a look at this even though I wasn't doing the ticket. As far as I can tell:  [`YearComparisonValidationModel`](https://github.com/publichealthengland/ntbs_Beta/blob/66dcb7ea50986e0274c2c5148c18ed59f134ef6e/ntbs-service/Models/Validations/NotificationEditValidationModels.cs#L34) doesn't allow for `NewYear` to be the empty string (it should be an `int`) so the request parsing magic just gives up and sets the whole request body to `null`, I think. Using `int?` probably would fix this, though it's worth noting that other Vue components in a similar situation will turn this empty string into a null before sending it (eg [`ValidateInput`](https://github.com/publichealthengland/ntbs_Beta/blob/66dcb7ea50986e0274c2c5148c18ed59f134ef6e/ntbs-service/wwwroot/source/Components/ValidateInput.ts#L28))

Reply:
Ah, that's interesting. Yes, I think we should use an `int?` so that we can just check whether `NewYear` is null here.

Reply:
As an aside, I'd much rather my parser return a `BadRequest` instead of silently setting the content to `null`, but I guess we can't do anything about that.